File: Elevador.py
import unittest

class TestElevador(unittest.TestCase):
    def setUp(self):
        self.elevador = Elevador(10, 0)

    def test_subir(self):
        self.elevador.subir()
        self.assertEqual(self.elevador.andar_atual, 1)

    def test_descer(self):
        self.elevador.subir()
        self.elevador.descer()
        self.assertEqual(self.elevador.andar_atual, 0)

if _name_ == "_main_":
    unittest.main()

class Elevador:
    def _init_(self, total_andares, andar_atual):
        self.total_andares = total_andares
        self.andar_atual = andar_atual

    def subir(self):
        if self.andar_atual < self.total_andares:
            self.andar_atual += 1

    def descer(self):
        if self.andar_atual > 0:
            self.andar_atual -= 1

File: Elevador.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class ElevadorTest {
    
    @Test
    void testSubir() {
        Elevador elevador = new Elevador(10, 0);
        elevador.subir();
        assertEquals(1, elevador.getAndarAtual());
    }

    @Test
    void testDescer() {
        Elevador elevador = new Elevador(10, 1);
        elevador.descer();
        assertEquals(0, elevador.getAndarAtual());
    }
}

public class Elevador {
    private int totalAndares;
    private int andarAtual;

    public Elevador(int totalAndares, int andarAtual) {
        this.totalAndares = totalAndares;
        this.andarAtual = andarAtual;
    }

    public void subir() {
        if (andarAtual < totalAndares) {
            andarAtual++;
        }
    }

    public void descer() {
        if (andarAtual > 0) {
            andarAtual--;
        }
    }

    public int getAndarAtual() {
        return andarAtual;
    }
}